### PR TITLE
Products API Endpoints

### PR DESCRIPTION
**Pull request by zstephenj**
Added Product and Category: Models, Controllers, API endpoints

The endpoints have some very minor validation placeholders, in no way meant to be the final server-side data validation!

AddProduct.vue refactored to turn image inputs into a base64 data string, image input errors added, refactored so form validation function is simpler

AddProduct.vue form bool properties refactored to be sent to the back-end as integer "1"s and "0"s 
      (bool was being changed in the SQL database to be of type "bit" (1s and 0s) I chose to change the property to integers so there does not have to be any conversion when we retrieve the data back from the database.)

.gitignore now ignoring Migrations/

### CODE REVIEW

**Review by ryanis-online — APPROVED** [review]
I feel like you would generally want to include the Migrations in the repo (especially given that an app that uses a trunk based release system would break if it weren't) but I'll approve anyways since I trust your judgement

**Comment by zstephenj** [comment]
No, I think you are right. I removed the reference to Migrations in our gitignore.

File: MooreMarket/Controllers/ProductsController.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        //PUT Products/{id}/Edit
+        [HttpPut("{id}/Edit")]
+        [ProducesResponseType(200)]
+        [ProducesResponseType(400)]
+        public IActionResult Edit(int id, Product product)
+        {
+            Product newProduct = product;
+            newProduct.ID = id;
+
+            if (newProduct.name == "")
```
Comment (iktani) [inline MooreMarket/Controllers/ProductsController.cs]:
```suggestion
            if (newProduct.Name == "")
```
VS Code complained about this line - lower case "n"ame

Comment (zstephenj) [reply MooreMarket/Controllers/ProductsController.cs]: Fixed it.

**Review by iktani — APPROVED** [review]
was able to get it working with your migrations folder after manually adding the migrations history table in. API endpoints all look good after testing in Postman. Well done!

**Comment by ryanis-online** [comment]
Looks great! Thanks 😄

**Review by cecoppinger — APPROVED** [review]
Cool seems to be working nice job.

**Review by simpson1986 — APPROVED** [review]

**Pull request merged**